### ISSUE TITLE
How can a plan be downgraded down to Free?

### ISSUE THREAD

**Issue by demoskp** [ohmybrew/laravel-shopify#116]
I tried having it as a plan costing 0 in my database but then that would not work as it tried to issue an application charge saying that there is a minimum charge for that and 0 is not a valid value.

 How would I go about cancelling the plan? Would I just issue a delete post request for the current charge instead of cancelling through the billingController?

**Comment by ohmybrew** [comment]
Yes double check the docs but I believe it's 0.50 min for a price on a application charge, recurring charge can be 0.

There's no cancel API, and because it's an error, Shopify shouldn't have made a charge.

**Comment by demoskp** [comment]
Still getting an error even with a recurring charge. I get the following error:

Client error: `POST https://domain.myshopify.com/admin/recurring_application_charges.json` resulted in a `422 Unprocessable Entity` response: {"errors":{"price":["must be greater than zero"]}}

**Comment by ohmybrew** [comment]
Can you show me what params you're sending. I just created a recurring plan
at 0.00 perfectly fine.

**Comment by demoskp** [comment]
I had price as 0 rather than 0.00. could that be the issue? I am not home now but I'll test it when I get home.

**Comment by ohmybrew** [comment]
I don't think so. You must supply capped_amount and terms if you wish to
set a recurring charge to 0

**Comment by ohmybrew** [comment]
@demoskp  https://github.com/ohmybrew/laravel-shopify/wiki/Creating-a-Billable-App#notes

**Comment by demoskp** [comment]
Yeah I missed that. What can the capped amount value be? Can it be set to 0?

**Comment by ohmybrew** [comment]
No you must set a maxiumum amount you would ever charge a store in a month. The Shopify docs has more information on this.

**Comment by fyroc** [comment]
> Yeah I missed that. What can the capped amount value be? Can it be set to 0?

Just make it the same value of your highest paid plan

**Comment by ncpope** [comment]
Deleting the charge is the right way to go: https://help.shopify.com/en/api/reference/billing/recurringapplicationcharge#destroy

Afterwards, you can delete the charge on your db manually. It's a soft delete so you can still reference if you need to in the future.

You will just need to make sure that your apps routes are not using the billable middleware otherwise the shop will be required to have an active recurring application charge to use your app.

**Comment by demoskp** [comment]
> 
> 
> Deleting the charge is the right way to go: https://help.shopify.com/en/api/reference/billing/recurringapplicationcharge#destroy
> 
> Afterwards, you can delete the charge on your db manually. It's a soft delete so you can still reference if you need to in the future.
> 
> You will just need to make sure that your apps routes are not using the billable middleware otherwise the shop will be required to have an active recurring application charge to use your app.

Do you mean not using the middleware for that specific route of downgrading the app to Free? Or do you mean all the routes?

**Comment by ohmybrew** [comment]
@demoskp To fix you're current issue, delete the application charge in the database to fix the previous mistake you had of the 0.00 with no cap and terms.

Then, create the recurring plan with 0.00, capped amount, and terms; then you should be good to roll.

**Comment by demoskp** [comment]
I implemented both ways and it works with both. Recurring plan 0.00 with capped amount works but I went with the more complicated route in the end and I am making a request to shopify to delete the current recurring application, soft deleting the recurring application from my database and setting it to inactive.

The only issue I faced with this is when getting the current recurring charge from the database to use it it a request to delete the recurring charge from shopify my value changes again from what it is in the database which I found is caused by the $casts in the charges model so I changed the charge_id cast from int to string and it now works.

**Comment by ohmybrew** [comment]
Ok good to note, ill update the cast!
